pkg/condition/condition.go

```diff
@@ -0,0 +1,267 @@
+package condition
+
+// adapted from rancher/wrangler
```

Comment:
Are there a large number of diffs vs wrangler?

Reply:
Only `GetLastTransitionTime` and `getLastTransitionTime` are new/different.